src/slug.ts

```diff
@@ -0,0 +1,19 @@
+export function slugref(envVar: string): string {
+  return envVar
+    .toLowerCase()
+    .replace(RegExp('refs/(heads|tags)/'), '')
+    .replace(new RegExp('[_/@]', 'g'), '-')
+    .substring(0, 63)
+}
+
+export function slugurl(envVar: string): string {
+  return slugref(envVar).replace(new RegExp('[.]', 'g'), '-')
+}
+
+export function slugurlref(envVar: string): string {
+  return slugurl(slugref(envVar))
+}
+
+export function shortsha(envVar: string): string {
+  return envVar.substring(0, 8)
```

Comment:
Those magic numbers should be put in constants which will add readability (with meaningful names), and maintainability (easy change of behaviour of your actions, or configuration).